src/pages/basic-concepts/content-markdown.md

```diff
@@ -3,7 +3,143 @@ layout: ~/layouts/Main.astro
+
+```jsx
+---
+// For now, this import _must_ be named "Markdown" and _must not_ be wrapped with a custom component
+// We're working on easing these restrictions!
+import { Markdown } from 'astro/components';
+import Layout from '../layouts/main.astro';
+import MyFancyCodePreview from '../components/MyFancyCodePreview.tsx';
+
+const expressions = 'Lorem ipsum';
+---
+
+<Layout>
+  <Markdown>
+    # Hello world!
+
+    **Everything** supported in a `.md` file is also supported here!
+
+    There is _zero_ runtime overhead.
+
+    In addition, Astro supports:
+    - Astro {expressions}
+    - Automatic indentation normalization
+    - Automatic escaping of expressions inside code blocks
+
+    ```jsx
+      // This content is not transformed!
+      const object = { someOtherValue };
+    ```
+
+    - Rich component support like any `.astro` file!
+    - Recursive Markdown support (Component children are also processed as Markdown)
+
+    <MyFancyCodePreview:visible>
+      ```jsx
+      const object = { someOtherValue };
+      ```
+    </MyFancyCodePreview:visible>
+  </Markdown>
+</Layout>
+```
+
+### Remote Markdown
+
+If you have Markdown in a remote source, you may pass it directly to the Markdown component through the `content` attribute. For example, the example below fetches the README from Snowpack's Github repository and renders it as HTML.
+
+```jsx
+---
+import { Markdown } from 'astro/components';
+
+const content = await fetch('https://raw.githubusercontent.com/snowpackjs/snowpack/main/README.md').then(res => res.text());
+---
+
+<Layout>
+  <Markdown content={content} />
+</Layout>
+```
+
+Some times you might want to combine dynamic markdown with static markdown. You can nest `Markdown` components to get the best of both worlds.
+
+```jsx
+---
+import { Markdown } from 'astro/components';
+
+const content = await fetch('https://raw.githubusercontent.com/snowpackjs/snowpack/main/README.md').then(res => res.text());
+---
+
+<Layout>
+  <Markdown>
+    ## Markdown example
+
+    Here we have some __Markdown__ code. We can also dynamically render content from remote places.
+
+    <Markdown content={content} />
+  </Mardown>
+</Layout>
+```
+
+### Security FAQs
+
+**Aren't there security concerns to rendering remote markdown directly to HTML?**
+
+Yes! Just like with regular HTML, improper use of the `Markdown` component can open you up to a [cross-site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting) attack. If you are rendering untrusted content, be sure to _sanitize your content **before** rendering it_.
+
+**Why not use a prop like React's `dangerouslySetInnerHTML={{ __html: content }}`?
+
+Rendering a string of HTML (or Markdown) is an extremely common use case when rendering a static site and you probably don't need the extra hoops to jump through. Rendering untrusted content is always dangerous! Be sure to _sanitize your content **before** rendering it_.
```

Comment:
should we have an appendix page to talk about sanitising content in astro, from a newbies perspective this could just sound like extra overhead that they might not be familiar with,

Reply:
This was great man, excellent work, =)

Reply:
Need @Hanawa02  to check it, and give it the stamp

Reply:
I think it's important to keep in the main page, since newbies *should* learn about it, though if we do end up having info on sanitizing content, maybe this section can stay the same, but with a call to action link to the sanitizing input page

Reply:
I think that's a good idea.

Reply:
Sorry, you are too fast (and that's good). Maybe I'm late, but I agree with @jasikpark. Will add the sanitizing page to one of the issue.